### PR TITLE
Install GitHub actions + fix the build

### PR DESCRIPTION
**Pull request by huitseeker**
## What this does
- implement Github actions running tests, integration tests, Clippy + rustfmt
- allow some clippy warnings to make the build pass,
- change a boundary condition in timeout checks in `client_conductor` to make the build pass

## Why 
Travis CI is slower at giving results (5-ish vs 16 min):
Selection_134
Selection_133

## Further work

Activate [caching of the aeronmd build](https://github.com/actions/cache) to save even more time.
 
 
 ## See Also
 
 Successful GH Action run on my fork [here](https://github.com/huitseeker/aeron-rs/actions/runs/603971323)
 
 /cc @denisxor @KiriosK who touched related files recently-ish

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  rustfmt:
+    name: rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Install minimal nightly with clipy
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
```
Comment (denisxor) [inline .github/workflows/main.yml:73]: Hi! Thanks for your contribution!
Is it possible to use GItHub actions with stabe Rust?

Comment (huitseeker) [reply .github/workflows/main.yml:73]: Sure! In fact, the *tests* job (starting line 8) is run with *both* stable and nightly (see the `${{matrix.toolchain}}`variable). This particular line only concerns the job running `clippy`. We can put `stable`, `nightly`, or anything else there. I'll add a commit switching this `clippy` run to stable, hoping that addresses your concern.

**Review by denisxor — APPROVED** [review]

File: src/image.rs [hunk 2]
```diff
@@ -41,6 +41,7 @@ use crate::utils::{
     types::Index,
 };
 
+#[cfg_attr(nightly, allow(clippy::upper_case_acronyms))]
```
Comment (KiriosK) [inline src/image.rs:44]: Thank you for contribution!
I just have a question about this clippy lint, which is used for nightly toolchain.
If we fix our CI to stable, as @denisxor mentioned, do we still need `rustc_version` and this one rule?

**Review by XX — APPROVED** [review]

File: src/image.rs (hunk 2 above)
Comment (huitseeker) [reply src/image.rs:44]: If your objective is to fix CI, then you're correct, you don't need to populate `rustc_version` or use this rule.

If your objective is to be able to pick up this code base and run a nightly version of `clippy` locally, you still do. Note that this could be useful, because [the nightly version allows developers to fix clippy lints automatically](https://github.com/rust-lang/rust-clippy#automatically-applying-clippy-suggestions).

Comment (KiriosK) [reply src/image.rs:44]: Didn't know about this one feature, can be useful for later changes.

**Review by KiriosK — APPROVED** [review]
